mlir/lib/Dialect/XeGPU/Transforms/XeGPUWgToSgDistribute.cpp

```diff
@@ -756,8 +738,9 @@ struct WgToSgArithConstantOp : public OpConversionPattern<arith::ConstantOp> {
     auto sgAttr = DenseElementsAttr::get(newType, singleVal);
     auto cstOp =
         arith::ConstantOp::create(rewriter, op.getLoc(), newType, sgAttr);
-    if (auto newLayout = layout.dropSgLayoutAndData())
-      xegpu::setDistributeLayoutAttr(cstOp->getResult(0), newLayout);
+    if (!layout.getLaneLayoutAsInt().empty())
```

Comment:
this check is not necessary, xegpu::setDistributeLayoutAttr(cstOp->getResult(0),layout.dropSgLayoutAndData()) is enough